interface/src/app/gene/searchFew/searchFew.js

```diff
@@ -51,17 +50,21 @@ angular.module('adage.gene.searchFew', [
               } else {
                 $scope.noResults = false;
               }
-
-              return data.results;
             },
 
             // Error function
             function(responseObject, responseHeaders) {
               $scope.errors = 'Gene search is temporarily down';
             }
-          ).$promise.then(function(response) {
-            return response.results;
-          });
+          ).$promise.then(
+            function(response) {
+              var selectedIDs = Object.keys($scope.selectedGenes);
+              var geneResultsList = response.results.filter(function(result) {
+                return selectedIDs.indexOf(result.id.toString()) === -1;
+              });
+              return geneResultsList.slice(0, numResultsToReturn);
+            }
+          );
```

Comment:
Agree that this is more like an enhancement. I think that setting `GENES_API_RESULT_LIMIT` to 20 will be probably enough for most use cases.